src/sentry/static/sentry/app/components/resultGrid.tsx

```diff
@@ -246,64 +268,61 @@ const ResultGrid = createReactClass({
         });
       },
     });
-  },
+  }
 
-  onSearch(e) {
-    const location = this.props.location || {};
+  onSearch(e: React.FormEvent<HTMLFormElement>) {
+    const location = this.props.location ?? {};
     const {query} = this.state;
-    const targetQueryParams = jQuery.extend({}, location.query || {}, {
-      query,
-      cursor: '',
-    });
+    const targetQueryParams = {...(location.query ?? {}), query, cursor: ''};
 
     e.preventDefault();
 
     browserHistory.push({
       pathname: this.props.path,
       query: targetQueryParams,
     });
-  },
+  }
 
-  onQueryChange(evt) {
+  onQueryChange(evt: React.ChangeEvent<HTMLInputElement>) {
     this.setState({query: evt.target.value});
-  },
+  }
 
   renderLoading() {
     return (
       <tr>
-        <td colSpan={this.props.columns.length}>
+        <td colSpan={this.props.columns!.length}>
           <div className="loading">
             <div className="loading-indicator" />
             <div className="loading-message">Hold on to your butts!</div>
           </div>
         </td>
       </tr>
     );
-  },
+  }
 
   renderError() {
     return (
       <tr>
-        <td colSpan={this.props.columns.length}>
+        <td colSpan={this.props.columns!.length}>
           <div className="alert-block alert-error">Something bad happened :(</div>
         </td>
       </tr>
     );
-  },
+  }
 
   renderNoResults() {
     return (
       <tr>
-        <td colSpan={this.props.columns.length}>No results found.</td>
+        <td colSpan={this.props.columns!.length}>No results found.</td>
```

Comment:
are you sure about the `!`? 
```suggestion
        <td colSpan={this.props.columns!.length}>{t('No results found.')}</td>
```